### ISSUE TITLE
Bootstrapping 4.0.0 fails

### ISSUE THREAD

**Issue by RainManreports** [easybuilders/easybuild-framework#3024]
Bootstrapping 4.0.0 currently fails under Fedora 30:

```
[rainman@localhost ~]$ EASYBUILD_PREFIX=$HOME/.local/easybuild
[rainman@localhost ~]$ python bootstrap_eb.py $EASYBUILD_PREFIX/
[[INFO]] EasyBuild bootstrap script (version 20190523.01, MD5: 92196312d8e88843ffe83f21a51f0592)
[[INFO]] Found Python 2.7.16 (default, Apr 30 2019, 15:54:43) ; [GCC 9.0.1 20190312 (Red Hat 9.0.1-0.10)]

[[INFO]] Installation prefix /home/rainman/.local/easybuild
[[INFO]] Found module command '/usr/share/lmod/lmod/libexec/lmod' via $LMOD_CMD (Lmod), so using it.
[[INFO]] Suitable setuptools installation already found, skipping stage 0...


[[INFO]] +++ STAGE 1: installing EasyBuild in temporary dir with easy_install...

[[INFO]] running pre-install command 'easy_install --quiet --upgrade --prefix=/tmp/tmpWcvj7D/eb_stage1 vsc-install<0.11.4'
[[INFO]] running pre-install command 'easy_install --quiet --upgrade --prefix=/tmp/tmpWcvj7D/eb_stage1 vsc-base<2.9.0'
[[INFO]] installing EasyBuild with 'easy_install --quiet --upgrade --prefix=/tmp/tmpWcvj7D/eb_stage1 easybuild'
  File "build/bdist.linux-x86_64/egg/easybuild/tools/py2vs3/py3.py", line 82
    class WrapperBase(object, metaclass=metaclass):
                                       ^
SyntaxError: invalid syntax

[[INFO]] running post install command 'easy_install --upgrade --prefix=/tmp/tmpWcvj7D/eb_stage1 vsc-base<2.9.0'


[[INFO]] +++ STAGE 2: installing EasyBuild in /home/rainman/.local/easybuild with EasyBuild from stage 1...

== temporary log file in case of crash /tmp/eb-6OMkjn/easybuild-m0BanK.log

WARNING: Use of 1 unknown easyconfig parameters detected in EasyBuild-4.0.0.eb: pyshortver
If these are just local variables please rename them to start with 'local_', or try using --fix-deprecated-easyconfigs to do this automatically.
For more information, see https://easybuild.readthedocs.io/en/latest/Easyconfig-files-local-variables.html .


WARNING: Deprecated functionality, will no longer work in v5.0: Use of 'dummy' toolchain is deprecated, use 'system' toolchain instead; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information

== processing EasyBuild easyconfig /tmp/tmpWcvj7D/EasyBuild-4.0.0.eb
== building and installing EasyBuild/4.0.0...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== restore after iterating...
== postprocessing...
== sanity checking...
== FAILED: Installation ended unsuccessfully (build directory: /tmp/tmpWcvj7D/EasyBuild/4.0.0/dummy-dummy): build failed (first 300 chars): Failed to find pattern '^\./easybuild_framework-4.0' in /home/rainman/.local/easybuild/software/EasyBuild/4.0.0/lib/python2.7/site-packages/easy-install.pth: ./vsc_install-0.11.3-py2.7.egg
 (took 26 sec)
== Results of the build can be found in the log file(s) /tmp/eb-6OMkjn/easybuild-EasyBuild-4.0.0-20190921.125647.oxRhZ.log
Traceback (most recent call last):
  File "bootstrap_eb.py", line 1101, in <module>
    main()
  File "bootstrap_eb.py", line 904, in main
    stage2(tmpdir, templates, install_path, distribute_egg_dir, sourcepath)
  File "bootstrap_eb.py", line 767, in stage2
    easybuild_main()
  File "/tmp/tmpWcvj7D/eb_stage1/lib/python2.7/site-packages/easybuild_framework-4.0.0-py2.7.egg/easybuild/main.py", line 434, in main
    ecs_with_res = build_and_install_software(ordered_ecs, init_session_state, exit_on_failure=exit_on_failure)
  File "/tmp/tmpWcvj7D/eb_stage1/lib/python2.7/site-packages/easybuild_framework-4.0.0-py2.7.egg/easybuild/main.py", line 147, in build_and_install_software
    raise EasyBuildError(test_msg)
easybuild.tools.build_log.EasyBuildError: 'Build of /tmp/tmpWcvj7D/EasyBuild-4.0.0.eb failed (err: "build failed (first 300 chars): Failed to find pattern \'^\\\\./easybuild_framework-4.0\' in /home/rainman/.local/easybuild/software/EasyBuild/4.0.0/lib/python2.7/site-packages/easy-install.pth: ./vsc_install-0.11.3-py2.7.egg\\n")
```
`

**Comment by boegel** [comment]
@RainManreports The bootstrap script is still being fixed, see https://github.com/easybuilders/easybuild-framework/pull/3017

Please try whether the updated bootstrap script from that PR works for you, and let us know.

You can download it as follows:
```
curl -OL https://raw.githubusercontent.com/easybuilders/easybuild-framework/e5c92697b7531a52943582b5d908bb1fee361386/easybuild/scripts/bootstrap_eb.py
```

You may see `SyntaxError: invalid syntax` during the installation for the `py2vs3/py3.py` module file when performing the bootstrap with Python 2, but you can ignore those safely, they will not affect the functionality of EasyBuild as long as you run it with Python 2 (see https://easybuild.readthedocs.io/en/latest/Python-2-3-compatibility.html).

I need to figure out how to conditionally include only either `py2vs3/py2.py` or `py2vs3/py3.py` based on the Python version used during the installation.

**Comment by boegel** [comment]
I've just updated PR #3017 to fix the problem @bartoldeman reported, after reproducing it.

@RainManreports To test the latest update to the bootstrap script (version `20190921.01`), please download it from https://raw.githubusercontent.com/easybuilders/easybuild-framework/eea990ce95283ef92391d319193825d339c7bc86/easybuild/scripts/bootstrap_eb.py

**Comment by RainManreports** [comment]
Works fine (bootstrapping EB 4.0.0 and building EB 3.9.4 with EB 4.0.0 just as a try). Thanks a lot!
Remaining issues as mentioned:
```
[[INFO]] installing EasyBuild with 'easy_install --quiet --upgrade --prefix=/tmp/tmpQV2OXm/eb_stage1 easybuild'
  File "build/bdist.linux-x86_64/egg/easybuild/tools/py2vs3/py3.py", line 82
    class WrapperBase(object, metaclass=metaclass):
                                       ^
SyntaxError: invalid syntax
```
`WARNING: Use of 1 unknown easyconfig parameters detected in EasyBuild-4.0.0.eb: pyshortver`

**Comment by boegel** [comment]
@RainManreports Perfect, thanks a lot for the feedback!

The warning about `pyshortver` is harmless too, that's due to the recommended naming scheme for local variables in easyconfig files (see https://easybuild.readthedocs.io/en/latest/Easyconfig-files-local-variables.html), which is easy to fix in the easyconfig file that the bootstrap scripts generates (I'll get that fixed in #3017 as well).

**Comment by bartoldeman** [comment]
Closed by  #3017
